source/blog/2022-11-17-announcing-hanami-200rc.html.markdown

```diff
@@ -0,0 +1,357 @@
+
+```ruby
+module Bookshelf
+  module Actions
+    module Books
+      class Show < Bookshelf::Action
+        include Deps["book_repo"]
+
+        params do
+          required(:slug).value(:string)
+        end
+
+        def handle(request, response)
+          book = book_repo.find_by_slug(request.params[:slug])
+
+          response.body = book.to_json
+        end
+      end
+    end
+  end
+end
+```
+
+Actions in Hanami still provide everything you need for a full-featured HTTP layer, including built-in parameter validation, Rack integration, session and cookie handling, flash messages, before/after callbacks and more.
+
+### Type-safe app settings
+
+For your app to do its thing, you'll want to give it various settings: behavioral toggles, API keys, external system URLs and the like. Hanami 2.0 provides a built-in settings class along with [dotenv][dotenv] integration for loading these settings from `.env*` files in local development.
+
+You can define your settings along with type constructors in `config/settings.rb`:
+
+```ruby
+module Bookshelf
+  class Settings < Hanami::Settings
+    setting :email_client_api_key, constructor: Types::String
+    setting :emails_enabled, constructor: Types::Params::Bool
+  end
+end
+```
+
+Your settings are then available as a `"settings"` component for you to use from anywhere in your app:
+
+```ruby
+# Given ENV["EMAILS_ENABLED"] = "true"
+Hanami.app["settings"].emails_enabled # => true
+```
+
+This means you can include `"settings"` in a list of `Deps` and access your settings from any class:
+
+```ruby
+module Bookshelf
+  module Emails
+    class DailyUpdate
+      include Deps["email_client", "settings"]
+
+      def deliver(recipient)
+        return unless settings.emails_enabled
+        # ...
+      end
+    end
+  end
+end
+```
+
+Hanami also loads your settings early in app boot and will raise an error for any settings that fail to meet their type expectations, giving you early feedback and ensuring your app doesn’t boot in a potentially invalid state.
+
+[dotenv]: https://github.com/bkeepers/dotenv
+
+### Providers
+
+So far we’ve seen how your classes and settings can be accessed as components in your app. But what about components that require special handling to initialize? For these, Hanami 2.0 introduces providers. Here’s one for the `"email_client"` component we used earlier, in `config/providers/email_client.rb`:
+
+```ruby
+Hanami.app.register_provider :email_client do
+  prepare do
+    require "acme_email/client"
+  end
+
+  start do
+    client = AcmeEmail::Client.new(
+      api_key: target["settings"].email_client_api_key,
+      default_from: "no-reply@bookshelf.example.com"
+    )
+
+    register "email_client", client
+  end
+end
+```
+
+Every provider has its own set of `prepare`/`start`/`stop` lifecycle steps, which is useful for setting up and using heavyweight components, or components that use tangible resources like database connections.
```

Comment:
```suggestion
Every provider has its own set of `prepare`/`start`/`stop` lifecycle steps, which is useful for setting up and using heavyweight components, or components that use tangible resources like database connections. Having a public API for managing component's **state** can be leveraged easily for things like rake tasks or CLI commands, where only a single component may be required. A common example of this would be `db:migrate` task or `assets:compile`, that can run in isolation from the rest of your application.
```

Reply:
Added this as a following paragraph, let me know what you think:

> You can interact directly with these providers and their components, giving you to access to parts of your app in situations where you may only need a single component. A common example here would be running database migrations or precompiling static assets, which are tasks that can run in isolation from the rest of your app.